power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -82,6 +84,35 @@ inline void update_y_bus(YBus<sym>& y_bus, std::shared_ptr<MathModelParam<sym> c
     y_bus.update_admittance(math_model_param);
 }
 
+template <bool sym>
+inline void update_y_bus(YBus<sym>& y_bus, std::shared_ptr<MathModelParam<sym> const> const& math_model_param,
+                         bool increment) {
+    auto branch_param_to_change_views =
+        std::views::iota(0, math_model_param->branch_param.size()) | std::views::filter([&math_model_param](Idx i) {
+            return math_model_param->branch_param[i].yff() != ComplexTensor<sym>{0.0} ||
+                   math_model_param->branch_param[i].yft() != ComplexTensor<sym>{0.0} ||
+                   math_model_param->branch_param[i].ytf() != ComplexTensor<sym>{0.0} ||
+                   math_model_param->branch_param[i].ytt() != ComplexTensor<sym>{0.0};
+        });
+    auto shunt_param_to_change_views =
+        std::views::iota(0, math_model_param->shunt_param.size()) | std::views::filter([&math_model_param](Idx i) {
+            return math_model_param->shunt_param[i] != ComplexTensor<sym>{0.0};
+        });
+
+    MathModelParamIncrement<sym> math_model_param_incrmt;
+    math_model_param_incrmt.branch_param = math_model_param->branch_param;
+    math_model_param_incrmt.shunt_param = math_model_param->shunt_param;
+    math_model_param_incrmt.source_param = math_model_param->source_param; // not sure if we actually need this
+    math_model_param_incrmt.branch_param_to_change =
+        std::vector<Idx>(branch_param_to_change_views.begin(), branch_param_to_change_views.end());
+    math_model_param_incrmt.shunt_param_to_change =
+        std::vector<Idx>(shunt_param_to_change_views.begin(), shunt_param_to_change_views.end());
```

Comment:
i think you can just do 
```suggestion
    math_model_param_incrmt.branch_param_to_change = branch_param_to_change_views;
    math_model_param_incrmt.shunt_param_to_change = shunt_param_to_change_views;
```

please do remember that this is a copy and is quite expensive. if possible, you might want to do an `std::move`